### PR TITLE
Couple of AbstractBlockState Mappings

### PR DESCRIPTION
**Pull request by dexman545**
AbstractBlock:
method_29291 -> toolRequired
field_25184 -> requiresTool

### CODE REVIEW

**Comment by Prospector** [comment]
I'd name the method requiresTool as well, the `does` prefix is quite odd imo.

**Comment by dexman545** [comment]
then the name would be the same as the setting's method that sets the block requiring a tool

**Comment by kashike** [comment]
`toolRequired`

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 1]
```diff
@@ -218,6 +218,7 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
 		FIELD field_23184 blockVisionPredicate Lnet/minecraft/class_4970$class_4973;
 		FIELD field_23185 postProcessPredicate Lnet/minecraft/class_4970$class_4973;
 		FIELD field_23186 emissiveLightingPredicate Lnet/minecraft/class_4970$class_4973;
+		FIELD field_25185 requiresTool Z
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		FIELD field_25185 toolRequired Z
```

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 2]
```diff
@@ -273,6 +274,7 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
 		METHOD method_26249 emissiveLighting (Lnet/minecraft/class_4970$class_4973;)Lnet/minecraft/class_4970$class_2251;
 			ARG 1 predicate
 		METHOD method_26250 air ()Lnet/minecraft/class_4970$class_2251;
+		METHOD method_29292 requiresTool ()V
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping:277]:
```suggestion
		METHOD method_29292 toolRequired ()V
```

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 3]
```diff
@@ -312,6 +314,7 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
 		FIELD field_23176 blockVisionPredicate Lnet/minecraft/class_4970$class_4973;
 		FIELD field_23177 postProcessPredicate Lnet/minecraft/class_4970$class_4973;
 		FIELD field_23178 emissiveLightingPredicate Lnet/minecraft/class_4970$class_4973;
+		FIELD field_25184 requiresTool Z
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		FIELD field_25184 toolRequired Z
```

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 4]
```diff
@@ -525,6 +528,7 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
 			ARG 2 pos
 		METHOD method_27852 isOf (Lnet/minecraft/class_2248;)Z
 			ARG 1 block
+		METHOD method_29291 toolRequired ()Z
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		METHOD method_29291 isToolRequired ()Z
```

**Review by liach — CHANGES_REQUESTED** [review]

**Comment by shedaniel** [comment]
I would prefer the old builder like names and requiresTool

**Comment by liach** [comment]
Usually we don't use third person singular tense in method names, so imo `requiresTool` isn't a good method name, especially for a setter. `requireTool` may be better, like we would declare `eat` than `eats` in a sheep class.

Also `toolRequired` is an adjective phrase, which is usually a better choice for names of boolean fields/variables.

**Comment by shedaniel** [comment]
then I would not use requireTool as the method, I can deal with toolRequired but please be consistent

**Comment by liach** [comment]
Sure, good point, updated my suggestion

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 2 above)
Comment (dexman545) [reply mappings/net/minecraft/block/AbstractBlock.mapping:277]: This change breaks with the other methods in setting's naming convention (being able to read the block settings in semi-natural English, ie. dropsNothing)

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 3 above)
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		FIELD field_25184 toolRequired Z
```
This is **not** something in the builder. You argue for following the style in the builder, but this is not builder and you should use the regular naming convention.

**Review by liach — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/block/AbstractBlock.mapping:277]: Mark this as resolved please.

**Comment by YanisBft** [comment]
It's a bit confusing the AbstractBlock.Settings field/method are different from the AbstractBlock.AbstractBlockState ones

**Comment by dexman545** [comment]
Woops, git messed up and had tried to push to the wrong place.
Field names should match now

**Review by YanisBft — APPROVED** [review]

**Comment by dexman545** [comment]
To help clarify:
(requiresTool)
```
public AbstractBlock.Settings method_29292() {
         this.field_25185 = true;
         return this;
}
```

is in the block settings, setting the value to true as part of a build-like creator, while (toolRequired)
```
public boolean method_29291() {
         return this.field_25184;
}
```
Gets the value

**Review by liach — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Pull request merged**